packages/babel-preset-env/src/index.ts

```diff
@@ -354,31 +337,34 @@ option \`forceAllTransforms: true\` instead.
   const exclude = transformIncludesAndExcludes(optionsExclude);
 
   const compatData = getPluginList(shippedProposals, bugfixes);
-  const shouldSkipExportNamespaceFrom =
-    (modules === "auto" && api.caller?.(supportsExportNamespaceFrom)) ||
-    (modules === false &&
-      !isRequired("transform-export-namespace-from", transformTargets, {
-        compatData,
-        includes: include.plugins,
-        excludes: exclude.plugins,
-      }));
-  const modulesPluginNames = getModulesPluginNames({
-    modules,
-    transformations: moduleTransformations,
-    // TODO: Remove the 'api.caller' check eventually. Just here to prevent
-    // unnecessary breakage in the short term for users on older betas/RCs.
-    shouldTransformESM: modules !== "auto" || !api.caller?.(supportsStaticESM),
-    shouldTransformDynamicImport:
-      modules !== "auto" || !api.caller?.(supportsDynamicImport),
-    shouldTransformExportNamespaceFrom: !shouldSkipExportNamespaceFrom,
-  });
+  const modules =
+    optionsModules === "auto"
+      ? api.caller(supportsStaticESM)
+        ? false
+        : "commonjs"
+      : optionsModules;
+  const shouldTransformDynamicImport =
+    optionsModules === "auto" ? !api.caller(supportsDynamicImport) : !!modules;
+
+  // If the caller does not support export-namespace-from, we forcefully add
+  // the plugin to `includes`.
+  // TODO(Babel 8): stop doing this, similarly to how we don't do this for any
+  // other plugin. We can consider adding bundlers as targets in the future,
+  // but we should not have a one-off special case for this plugin.
+  if (
+    optionsModules === "auto" &&
+    !api.caller(supportsExportNamespaceFrom) &&
+    !exclude.plugins.has("transform-export-namespace-from")
+  ) {
+    include.plugins.add("transform-export-namespace-from");
+  }
```

Comment:
@JLHwung Now `transform-export-namespace-from` is 100% handled as a normal plugin, with this `if` statement being the only exception.

The other two modules plugins are still "special" because it does make no sense to do targets-based compilation for them given that they completely change the underlying module system, so they are still handled by the separate function.

If you look at the diffs, now the only actual output difference is that we enable `syntax-dynamic-import` in some cases where we weren't doing so previously.